akka-http/src/main/scala/akka/http/javadsl/server/directives/HeaderDirectives.scala

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (C) 2009-2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.http.javadsl.server.directives
+
+import java.util.Optional
+import java.util.{function => jf}
+
+import scala.compat.java8.OptionConverters._
+import scala.reflect.ClassTag
+
+import akka.http.javadsl.model.HttpHeader
+import akka.http.javadsl.server.Route
+import akka.http.scaladsl.server.directives.{HeaderDirectives => D}
+import akka.http.scaladsl.server.util.ClassMagnet
+
+/**
+ * 
+ * 
+ * FIXME When implementing custom headers in Java, just extend akka.http.scaladsl.model.headers.CustomHeader,
+ * since the java API's CustomHeader class does not implement render().
```

Comment:
Sorry, I'm blind - we can't because of the path dependent type in the render signature... 
/me thinking about it (could be solved after the initial PR IMO)